File: shared/types/agent.ts
export type AgentRole = 'moderator' | 'scribe' | 'participant';
export type NoveltyTier = 'first_argument' | 'correction' | 'new_information' | 'disagreement' | 'synthesis' | 'reinforcement';

export interface Agent {
    id: string;
    session_id: string;
    display_name: string;
    persona_description?: string;
    expertise?: string;
    llm_provider: string;
    llm_model: string;
    llm_config?: Record<string, any>;
    role: AgentRole;
}

export interface AgentPreset {
    id: string;
    display_name: string;
    persona_description: string;
    expertise: string;
    suggested_model: string;
    llm_provider: string;
    category: string;
    is_system: boolean;
}

export interface QueueEntry {
    agent_id: string;
    agent_name?: string;
    priority_score: number;
    novelty_tier: NoveltyTier;
    justification?: string;
    position: number;
}

File: backend/schemas/agent.py
from pydantic import BaseModel
from typing import Dict, Literal, Optional

AgentRole = Literal['moderator', 'scribe', 'participant']
NoveltyTier = Literal[
    'first_argument', 'correction', 'new_information',
    'disagreement', 'synthesis', 'reinforcement',
]

class AgentSchema(BaseModel):
    id: str
    session_id: str
    display_name: str
    persona_description: Optional[str] = None
    expertise: Optional[str] = None
    llm_provider: str
    llm_model: str
    llm_config: Optional[Dict] = None
    role: AgentRole

class AgentPresetSchema(BaseModel):
    id: str
    display_name: str
    persona_description: str
    expertise: str
    suggested_model: str

class QueueEntrySchema(BaseModel):
    agent_id: str
    agent_name: Optional[str] = None
    priority_score: float
    novelty_tier: NoveltyTier
    justification: Optional[str] = None
    position: int
